### DISCUSSION TITLE
Mismatch of current selection between EVCC itself and this integration?

### DISCUSSION THREAD

**Discussion by MortuuS** [marq24/ha-evcc#3] [General]
Hi!
First of all thanks for this integration. But I think I found a bug, or I am misunderstanding something?

In the dropdown menu of 'charging current max' the selections range from 0.125A to **14A**. While the dropdown menu in EVCC goes from 6A to 32A. 
Similar problem with 'charging current min' this integration goes from 6A to 32A. While EVCC goes from 0.125A to **16A**
So these inputs seem switched to me? And also the upper limit for minimum current seems incorrect (14A vs 16A).

![Screenshot 2024-06-03 172958](https://github.com/marq24/ha-evcc/assets/6415298/e3848bd9-19e0-4546-ae39-1452c6de765e)
![Screenshot 2024-06-03 173031](https://github.com/marq24/ha-evcc/assets/6415298/defa091c-52df-434d-a465-3d5ec68ce703)

Thanks again. 👍

**Comment by marq24** [comment]
Thanks for your report - indeed a BUG that was introduced with the adjustments of #2

**Comment by marq24** [comment]
the build 2024.6.0 will solve the issue - the options should be correct now - if you agree, that the new version solves the issue it would be cool, if you could close the bug #4

**Reply by MortuuS** [reply]
Build 2024.6.0 indeed fixes the issues I mentioned. I also replied to the issue that was created. But since you created it from my 'discussion' post. I'm unable to close the issue as far as I can see.
Thanks for the quick fix. :)

**Comment by robbedenys** [comment]
Hi, sorry to revive this old topic, but I think it's the closest match. It looks like the options of the charging current min select are hard-coded rather than derived from the evcc configuration. Running ha-evcc 2026.8.5: my charger and vehicle (Ioniq 5) are configured in evcc for single-phase charging from 6 A up to 32 A, and evcc itself happily accepts a min current of 20 A via its UI/API - but the integration's min-current select stops at 16 A (0.125 A … 16 A), while the max-current select goes up to 64 A. Reloading the integration doesn't change the option list.

Use case: single-phase grid charging on cheap tariff slots, where the minimum current is the actual charging setpoint (in min+pv mode), so values above 16 A are legitimate on a 32 A single-phase charger.

Would it be possible to derive the min-current options from the loadpoint's/vehicle's maxCurrent (or simply extend the list to 32 A, capped by the configured max)? Right now I work around it with a rest_command to POST /api/loadpoints/1/mincurrent/{amps}. Thanks for the great integration!

**Reply by marq24** [reply]
I am sorry - but I am not able to follow you here...

When a Loadpoint have a min and a max current option in evcc, than both of the options are available in HA-evcc integration...

And the range from the min current depends on the current value of the max-current - and vise versa...  - You might like to create some screenshots where you show a) what you have in evcc - and what you get in HA... 

And then try to explain, what you want to have? [for sure the Integration will only allow settings/values that are also available in evcc]

**Comment by robbedenys** [comment]
Hi, thanks for your answer. 

This is the current setting in EVCC:

<img width="690" height="138" alt="Schermafbeelding 2026-08-19 084859" src="https://github.com/user-attachments/assets/4f7f0bf7-180f-4f29-a2cf-872a802a0c3f" />

I can choose 20A min current in EVCC:

<img width="750" height="348" alt="Schermafbeelding 2026-08-18 143248" src="https://github.com/user-attachments/assets/eb4e0577-34af-4d04-aafa-4530cf7d06d8" />

Here are all the options in EVCC for the min current dropdown:

<img width="731" height="1512" alt="Schermafbeelding 2026-08-19 084806" src="https://github.com/user-attachments/assets/b2f02b7a-5663-4d71-9abf-d62077aa8609" />

These are the options I have in HA-evcc (eve n after doing a reload of the integration):

<img width="1026" height="483" alt="Schermafbeelding 2026-08-19 084836" src="https://github.com/user-attachments/assets/b56c722b-4172-4532-af69-b2e666ed98dc" />

So I can select 20A min current (max current is 32A) in EVCC for example, but I cannot choose this in HA-evcc.
Hope this clears it up.

**Reply by marq24** [reply]
thanks _now_ I got/understood the issue - and this looks like a new bug

**Reply by marq24** [reply]
@robbedenys please update to https://github.com/marq24/ha-evcc/releases/tag/2026.8.8

**Reply by robbedenys** [reply]
Thanks!! It is fixed now! :-)
